src/main/java/net/mcreator/ui/MCreator.java

```diff
@@ -288,24 +289,27 @@ else if (screenSize.getWidth() > 1290 && screenSize.getHeight() > 795)
 			// backup if new version and backups are enabled
 			if (workspace.getMCreatorVersion() < Launcher.version.versionlong
 					&& PreferencesManager.PREFERENCES.backups.backupOnVersionSwitch) {
-				ShareableZIPManager.exportZIP(L10N.t("dialog.workspace.export_backup"),
-						new File(workspace.getFolderManager().getWorkspaceCacheDir(),
-								"FullBackup" + workspace.getMCreatorVersion() + ".zip"), this, true);
+				ThreadUtil.runOnSwingThreadAndWait(

```

Comment:
Those calls may not be needed as setVisible is/should be called from UI thread anyway (test this)